dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/enrollment/AbstractEnrollmentService.java

```diff
@@ -288,27 +290,20 @@ public List<Enrollment> getEnrollments( Iterable<ProgramInstance> programInstanc
     }
 
     @Override
-    public Enrollment getEnrollment( String id )
+    public Enrollment getEnrollment( String id, InstanceParams params )
     {
-        ProgramInstance programInstance = programInstanceService.getProgramInstance( id );
-        return programInstance != null ? getEnrollment( programInstance ) : null;
+        return Optional.ofNullable( programInstanceService.getProgramInstance( id ) )
+            .map( pi -> getEnrollment( pi, params ) ).orElse( null );
     }
 
     @Override
-    public Enrollment getEnrollment( ProgramInstance programInstance )
-    {
-        return getEnrollment( currentUserService.getCurrentUser(), programInstance, TrackedEntityInstanceParams.FALSE,
-            false );
-    }
-
-    @Override
-    public Enrollment getEnrollment( ProgramInstance programInstance, TrackedEntityInstanceParams params )
+    public Enrollment getEnrollment( ProgramInstance programInstance, InstanceParams params )
```

Comment:
Why was `getEnrollment( ProgramInstance programInstance )` removed?
I can see that in other classes we are calling a lot of times `getEnrollment( programInstance, TrackedEntityInstanceParams.FALSE )`.
Wouldn't make sense to maintain one method without the params that are default to FALSE or we would change some behavior that I cannot see